File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    if(s[0] == ")"): return False
    stack = []
    for x in s:
    	if(x == "("): 
            stack.append(x)
    	else:
            if(len(stack) != 0) :
            	stack.pop()
    return len(stack) == 0

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    // stack for checking
    const stack = [];
    
    for(const c of s){
        
        // when stack is empty
        if(stack.length < 1){
            if(c === "("){
            	console.log("Push (")
        		stack.push(c);
            } else {
                return false;
            }
            continue;
        }
        
        // when stack has item
        if(c === "(") {
            stack.push(c);
        } else {
            // c === ")"
            const lastIndex = stack.length - 1;
            const top = stack[lastIndex];
            if(top === "(") {
                stack.pop();
            }
        }
    }
    

    return stack.length === 0;
}